### PR TITLE
Cleanup and optimizations

### PR DESCRIPTION
**Pull request by lucaslorentz**
- Optimizations
- Removed unused methods, classes, properties and so on.
- Used JSON serialization attributes to reduce our serialization related code.

I still need to update tests.

Can you please check if you get good performance on your test projects?

### CODE REVIEW

File: src/MiniCover/HitServices/HitTestMethod.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            MethodName = methodName;
+            AssemblyLocation = assemblyLocation;
+            Counter = counter;
+        }
+
+        public string AssemblyName { get; }
+        public string ClassName { get; }
+        public string MethodName { get; }
+        public string AssemblyLocation { get; }
+        public int Counter { get; set; }
```
Comment (bhugot) [inline src/MiniCover/HitServices/HitTestMethod.cs]: why public set?

File: src/MiniCover/HitServices/Hits.cs [hunk 2]
```diff
@@ -1,25 +1,19 @@
 ﻿using MiniCover.Utils;
```
Comment (bhugot) [inline src/MiniCover/HitServices/Hits.cs]: remove unused using

**Review by bhugot — CHANGES_REQUESTED** [review]

**Comment by bhugot** [comment]
I m going to merge this after.
I still see 2 problems:
Using newtonsoft.json is nice but we can be overriding the one from a project. we need to solve that.
Like I said I think we could have a better performance if we take as start point test method and not Hit, and that would allow us to save after each test

File: src/MiniCover/HitServices/HitTestMethod.cs (hunk 1 above)
Comment (lucaslorentz) [reply src/MiniCover/HitServices/HitTestMethod.cs]: Fixed

File: src/MiniCover/HitServices/Hits.cs (hunk 2 above)
Comment (lucaslorentz) [reply src/MiniCover/HitServices/Hits.cs]: Fixed

**Review by bhugot — APPROVED** [review]

**Comment by lucaslorentz** [comment]
I don't think the performance will increase a lot if we take as start point the test method.
We already removed most O(n) operations during hit.
The only O(n) operation now is getting the test method, but it is executed only on top methods (not nested ones).

**Comment by lucaslorentz** [comment]
Also, on this PR I removed extra conversions we was doing from MethodBase to TestMethodInfo.
Before we was converting every time, and also cloning it.

**Comment by bhugot** [comment]
I m more think for saving process at the end 
If each time test change we save the test and his hits. we should be fine.

**Comment by bhugot** [comment]
oh you removed the cloning? then the count can be wrong

**Comment by lucaslorentz** [comment]
No, it's not. Because I changed when the TestMethodInfo is created.

**Comment by bhugot** [comment]
oh ok

**Pull request merged**